File: backend/app/schemas/audit.py
from datetime import datetime

from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    id: str
    company_id: str
    user_id: str | None
    user_name: str | None = None
    action: str
    entity_type: str
    entity_id: str | None
    changes: dict | None = None
    ip_address: str | None
    user_agent: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class AuditLogListResponse(BaseModel):
    items: list[AuditLogResponse]
    total: int
    page: int
    per_page: int

File: frontend/src/types/audit.ts
export interface AuditLogEntry {
  id: string;
  company_id: string;
  user_id: string | null;
  user_name: string | null;
  action: string;
  entity_type: string;
  entity_id: string | null;
  changes: Record<string, { old: unknown; new: unknown }> | null;
  ip_address: string | null;
  user_agent: string | null;
  created_at: string;
}

export interface PaginatedAuditLogs {
  items: AuditLogEntry[];
  total: number;
  page: number;
  per_page: number;
}

export interface AuditLogFilters {
  user_id?: string;
  action?: string;
  entity_type?: string;
  entity_id?: string;
  date_from?: string;
  date_to?: string;
}
